### PR TITLE
[SMALLFIX] Merging branch 1.2 back to master

### PR DESCRIPTION
**Pull request by jsimsa**

### CODE REVIEW

**Comment by jsimsa** [comment]
Jenkins, test this please

File: core/common/src/test/java/alluxio/heartbeat/SleepingTimerTest.java [hunk 1]
```diff
@@ -53,6 +53,7 @@ public void warnWhenExecutionTakesLongerThanInterval() throws Exception {
   public void sleepForSpecifiedInterval() throws Exception {
     final SleepingTimer timer =
         new SleepingTimer(THREAD_NAME, INTERVAL_MS, mMockLogger, mFakeClock, mMockSleeper);
+    timer.tick(); // first tick won't sleep
```
Comment (calvinjia) [inline core/common/src/test/java/alluxio/heartbeat/SleepingTimerTest.java]: Maybe add a check after to verify it did not sleep?

Comment (jsimsa) [inline core/common/src/test/java/alluxio/heartbeat/SleepingTimerTest.java]: done

**Comment by jsimsa** [comment]
@calvinjia recorded the conflicts first and then resolved them in separate commits, PTAL

**Comment by jsimsa** [comment]
@calvinjia FYI, I added the wrapper PR to this merge

**Comment by apc999** [comment]
LGTM

**Comment by apc999** [comment]
@calvinjia , is this PR looking good to you?

**Comment by calvinjia** [comment]
LGTM

**Pull request merged**